packages/lab/src/date-picker/DatePicker.tsx

```diff
@@ -34,21 +38,17 @@ export interface DatePickerProps
    */
   disabled?: boolean;
   /**
-   * The start date value. Use when the component is controlled.
-   */
-  startDate?: DateValue;
-  /**
-   * The default start date value. Use when the component is not controlled.
+   * The selected date value. Use when the component is controlled.
+   * Can be a single date or an object with start and end dates for range selection.
    */
-  defaultStartDate?: DateValue;
+  selectedDate?: DateValue | { startDate: DateValue; endDate: DateValue };
```

Comment:
This needs changeset

[good for future] look into generics to align prop types